example/lib/example_basemap_gallery.dart

```diff
@@ -0,0 +1,182 @@
+    )..setViewStyle(BasemapGalleryViewStyle.grid);
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('BasemapGallery'),
+        actions: [
+          IconButton(
+            tooltip: 'Show basemap gallery',
+            icon: const Icon(Icons.layers_outlined),
+            onPressed: _showBasemapGallerySheet,
+          ),
+        ],
+      ),
+      body: _buildMapPane(),
+    );
+  }
+
+  Widget _buildMapPane() {
+    return ArcGISMapView(
+      controllerProvider: () => _mapViewController,
+      onMapViewReady: () {
+        _mapViewController.arcGISMap = _map;
+      },
+    );
+  }
+
+  List<BasemapGalleryItem> _makeBasemapGalleryItems() {
+    const identifiers = <String>[
+      '46a87c20f09e4fc48fa3c38081e0cae6',
+      'f33a34de3a294590ab48f246e99958c9',
+      '52bdc7ab7fb044d98add148764eaa30a', // mismatched spatial reference
+      '3a8d410a4a034a2ba9738bb0860d68c4', // incorrect portal item type
+    ];
+
+    return identifiers
+        .map((id) {
+          final uri = Uri.parse('https://www.arcgis.com/home/item.html?id=$id');
+          final portalItem = PortalItem.withUri(uri);
+          if (portalItem == null) return null;
+          return BasemapGalleryItem(basemap: Basemap.withItem(portalItem));
+        })
+        .whereType<BasemapGalleryItem>()
+        .toList(growable: false);
+  }
+
+  Future<void> _showBasemapGallerySheet() async {
+    await showModalBottomSheet<void>(
+      context: context,
+      useSafeArea: true,
+      isScrollControlled: true,
+      builder: (context) {
+        return FractionallySizedBox(
+          heightFactor: 0.75,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              Padding(
+                padding: const EdgeInsets.fromLTRB(12, 12, 12, 6),
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: ValueListenableBuilder<BasemapGalleryItem?>(
+                        valueListenable: _controller.currentBasemapNotifier,
+                        builder: (context, currentItem, _) {
+                          final name = currentItem?.name ?? '';
+                          return Text(
+                            name.isEmpty
+                                ? 'Select a basemap'
+                                : 'Selected: $name',
+                            maxLines: 1,
+                            overflow: TextOverflow.ellipsis,
+                            style: Theme.of(context).textTheme.titleMedium,
+                          );
+                        },
+                      ),
+                    ),
+                    TextButton(
+                      onPressed: () => Navigator.of(context).pop(),
+                      child: const Text('Done'),
```

Comment:
It is weird to have a 'done' text button here,  Could you change to an icon button (x)